### DISCUSSION TITLE
Single JSON Object Consumption

### DISCUSSION THREAD

**Discussion by Ron-Gowen** [CMSgov/price-transparency-guide#600] [Q&A]
Our team has begun looking to consume these files and create tools around them. We are running into some serious challenges in regards to the file size and the fact that the files are basically one large JSON object all on one line. The memory required to read these files is proving to be a roadblock to success. The files can sometimes approach a TB in size. 

Can I ask what tools are other developers using to get past the size issues and be able to load these files for proper processing?

**Comment by shaselton-usds** [comment]
For larger files, don't read the contents into memory. Stream the input for parsing. This is a pretty common practice.

**Comment by liuj4** [comment]
See this thread for earlier discussion of this topic.
https://github.com/CMSgov/price-transparency-guide/discussions/528

**Comment by dgolden** [comment]
@Ron-Gowen Consuming these MRF is a substantial effort.  My company www.keyark.com handles these, but that's because we invested 3 years in creating a general purpose platform for processing complex, non-flat data. We are available to assist organizations that need the MRF data provided in a simplified, flattened form. 
See discussion #590 for a screenshot of our tool.

**Comment by harithatavarthy** [comment]
Take a look at this blog post: https://aws.amazon.com/blogs/big-data/process-price-transparency-data-using-aws-glue/
It talks about splitting large JSON file into smaller chunks before it can be processed and analyzed.
While the blog focuses on AWS Glue for preprocessing and processing the data, ideally it can be processed using any python/spark platform
